Readme.md

```diff
@@ -554,11 +554,13 @@ program.on('option:verbose', function () {
   process.env.VERBOSE = this.verbose;
 });
 
-// custom error on unknown command
 program.on('command:*', function (operands) {
-  console.error(`Invalid command '${operands[0]}'. Did you mean:`);
-  console.error(mySuggestions(operands[0]));
-  process.exit(1);
+  console.error(`error: unknown command '${operands[0]}'`);
+  const availableCommands = program.commands.map(cmd => cmd.name());
+  const suggestion = didYouMean(operands[0], availableCommands);
+  if (suggestion)
+    console.error(`Did you mean '${suggestion}'?`);
```

Comment:
This would be misleading. Say for instance, a first time user makes use of this code snippet to handle unknown commands ending up with an error saying that `didYouMean` is not defined.

Reply:
An alternative I considered was to use a clearly write-your-own routine like:

```
const availableCommands = program.commands.map(cmd => cmd.name());
const suggestion = myFindBestMatch(operands[0], availableCommands);
console.error(`Did you mean '${suggestion}'?`);
process.exitCode = 1;
```

Do you think this would be preferable?

Reply:
https://github.com/tj/commander.js/issues/1015#issuecomment-583810519